### DISCUSSION TITLE
Safe to expose to the internet?

### DISCUSSION THREAD

**Discussion by ollytom** [ngoduykhanh/PowerDNS-Admin#1250] [Q&A] [answered]
My friend works at a company which uses this software to adminster ~500 zones.
We were chatting and wondering if this software is safe to expose to the internet, fronted with a HTTP reverse proxy like nginx.

Currently, at their company, it is available only through VPN. After connecting to the VPN, they authenticate to pdns-admin via SAML; 2FA is enabled.

What are the community's thoughts and/or experience of exposing pdns-admin to the internet?
Thanks in advance!

**Comment by jbe-dw** [comment, marked as answer]
Hello,

Currently PDA has no enterprise grade code. It's a personal development shared to the community. We don't have a sufficient involvement to improve this quickly. Due to this lack of quality, I strongly advise to not expose enterprise DNS data to the internet. Someone could temper your DNS to do malicious things or have an easy access to your infrastructure's address plan.

You could add safeguards like ip filtering on the proxy, but it's better to use a VPN connection. Any DNS entry that should be viewed can be fetched from your DNS servers, and edition should be opened to a limited subset of people only.

For safety it should remain internal. Any help on improving and enforcing software's security is welcome though !

Regards
